framework/Optimizers/Optimizer.py

```diff
@@ -288,7 +288,9 @@ def _readMoreXMLbase(self,xmlNode):
         if self.fullOptVars is None:
           self.fullOptVars = []
         # store variable name
-        varName = child.parameterValues['name']
+        # Added by alfoa: the variable name is always considered a single string.
+        #                 If a comma is present, we remove any leading spaces here
+        varName = ','.join([key.strip() for key in child.parameterValues['name'].split(",")])
```

Comment:
It seems to me, we may have a different string type in the InputData to handle this exception.

Reply:
That might be true; however, to be compatible with the current algorithms you'd have to check for length 1 arrays and collapse them into strings if it's a single variable, so I'm not sure there's a big gain there.

Reply:
Why do we need to check the length 1 arrays? We treat the user provided name as string, the question is do we want to remove the whitespace or not inside the string when comma is in the string.

Reply:
The reason I kept it in here just removing the leading whitespaces in case of a comma is because we treat the varNames a single entities...For example for adaptive samplers (dimensions). I can modify the "name" attribute to be a comma separated list and not a StringType but I need to parse all the samplers to make sure we do not miss any option.

Let me know what you want me to do.

Reply:
After talking with @wangcj05, I got his point. I addressed his comment with new InputData type